server/src/main/java/org/opensearch/index/store/remote/utils/TransferManager.java

```diff
@@ -85,15 +87,23 @@ private static FileCachedIndexInput createIndexInput(FileCache fileCache, BlobCo
             try {
                 if (Files.exists(request.getFilePath()) == false) {
                     try (
-                        InputStream snapshotFileInputStream = blobContainer.readBlob(
-                            request.getBlobName(),
-                            request.getPosition(),
-                            request.getLength()
-                        );
+                        // Create a new empty file in the specified path
                         OutputStream fileOutputStream = Files.newOutputStream(request.getFilePath());
-                        OutputStream localFileOutputStream = new BufferedOutputStream(fileOutputStream)
+
+                        // All blob parts will be appended to file iteratively.
+                        OutputStream localFileOutputStream = Files.newOutputStream(request.getFilePath(), APPEND)
```

Comment:
```suggestion
                        OutputStream localFileOutputStream = new BufferedOutputStream(fileOutputStream)
```

Why doesn't this work? You're missing the buffered output stream in your code, plus I don't see any reason to reopen the file?

Reply:
Sorry while testing I was putting the outstream inside loop and seen overriding in every loop. To overcome this overriding I was using APPEND but when I removed outputstream out of  loop this overriding doesn't happen anymore.

Yes you are right your suggestion is working. Will update the PR in a few